notifications_utils/clients/statsd/statsd_client.py

```diff
@@ -9,10 +19,32 @@ def __init__(self, host, port, prefix):
         self._port = port
         self._prefix = prefix
         self._sock = socket(AF_INET, SOCK_DGRAM)
+        self._queue = deque()
+
+    def _resolve(self, addr):
+        return gethostbyname(addr)
+
+    @cachetools.func.ttl_cache(maxsize=2, ttl=10, timer=time_monotonic_with_jitter)
```

Comment:
TTL of 10 with jitter, meaning this will be cached between 7-13s.

Reply:
Why did you opt for this approach as opposed to just using the 15s retry that the UK team did? Not that it's a problem just curious.

Reply:
The UK team has a 15s retry with jitter. I switched to 10s with jitter because I think 10s is enough to retry resolving the host and we're adding points to a queue and I don't want it to grow too much.